src/locale/uk.js

```diff
@@ -28,39 +28,53 @@ function relativeTimeWithPlural(number, withoutSuffix, key) {
         return number + ' ' + plural(format[key], +number);
     }
 }
+var weekdays = {
+    nominative: 'неділя_понеділок_вівторок_середа_четвер_п’ятниця_субота'.split('_'),
+    accusative: 'неділю_понеділок_вівторок_середу_четвер_п’ятницю_суботу'.split('_'),
+    genitive: 'неділі_понеділка_вівторка_середи_четверга_п’ятниці_суботи'.split('_')
+};
 function weekdaysCaseReplace(m, format) {
-    var weekdays = {
-        'nominative': 'неділя_понеділок_вівторок_середа_четвер_п’ятниця_субота'.split('_'),
-        'accusative': 'неділю_понеділок_вівторок_середу_четвер_п’ятницю_суботу'.split('_'),
-        'genitive': 'неділі_понеділка_вівторка_середи_четверга_п’ятниці_суботи'.split('_')
-    };
-
     if (!m) {
         return weekdays['nominative'];
     }
 
     var nounCase = (/(\[[ВвУу]\]) ?dddd/).test(format) ?
         'accusative' :
-        ((/\[?(?:минулої|наступної)? ?\] ?dddd/).test(format) ?
+        ((/\] ?dddd/).test(format) ?
             'genitive' :
             'nominative');
     return weekdays[nounCase][m.day()];
 }
 function processHoursFunction(str) {
     return function () {
-        return str + 'о' + (this.hours() === 11 ? 'б' : '') + '] LT';
+        return str + (this.hours() === 11 ? 'об' : 'о') + '] LT';
     };
 }
+var monthsParse = [/^січ/i, /^лют/i, /^бер/i, /^кві/i, /^тра/i, /^чер/i, /^лип/i, /^сер/i, /^вер/i, /^жов/i, /^лис/i, /^гру/i];
+
+var monthsPattern = '(?:січ|берез|квіт|трав|черв|лип|серп|верес|жовт|груд)(?:ень|ня)|лют(?:ий|ого)|листопада?';
+var monthsShortPattern = '(?:січ|лют|бер(?:ез)?|квіт?|трав?|черв?|лип|серп?|вер(?:ес)?|жовт?|лис(?:т(?:оп)?)?|груд?)\\.?';
```

Comment:
Where did you take those short names from? Please see the discussion in #1001.

Reply:
From this standard https://www.uni-sport.edu.ua/sites/default/files/vseDocumenti/dokument2.pdf